pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2021 ScyllaDB
+
+package operator
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+)
+
+func TestMain(m *testing.M) {
+	dynamiccertificates.FileRefreshDuration = 1 * time.Second
+	os.Exit(m.Run())
+}
+
+func TestRunWebhookServerCertReload(t *testing.T) {
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir := t.TempDir()
+	TLSCertFile := dir + "/tls.crt"
+	TLSKeyFile := dir + "/tls.key"
+
+	cc, err := newCertificateCreator()
+	if err != nil {
+		t.Fatalf("can't create CertificateCreator: %v", err)
+	}
+
+	certSerialNumber := big.NewInt(1)
+	encodedCert, err := cc.newEncodedCertificate(certSerialNumber)
+	if err != nil {
+		t.Fatalf("can't create new encoded certificate: %v", err)
+	}
+	if err = createFileWithContent(TLSCertFile, encodedCert.cert); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+	if err = createFileWithContent(TLSKeyFile, encodedCert.privateKey); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+
+	wo := WebhookOptions{
+		TLSCertFile:                    TLSCertFile,
+		TLSKeyFile:                     TLSKeyFile,
+		Port:                           5000,
```

Comment:
a fixed port might be already taken on a machine, let's use a dynamic one with `:0` and detect which was allocated